### PR TITLE
Fix: Error on getting event detail by identifier

### LINKED ISSUE #3976: Unable to request events using identifier.
When requesting events using their identifier, it results in `500 Internal Server Error`. 

Eg: 
- https://api.eventyay.com/v1/events/b70d9fd1 
- https://open-event-api.herokuapp.com/v1/events/3d07e4f5

Please ensure such major regressions don't occur in future. It hard to work on clients, when such a major endpoint fails.

### PR DESCRIPTION
**Pull request by enigmaeth**
Fixes #3976 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `nextgen` branch.
- [x] The unit tests pass locally with my changes 

#### Short description of what this resolves:
Fixes Error on getting event detail by identifier

(sorry it was assigned to @SaptakS, but I made the PR because it's a high-priority issue and was open for 2 days)

### CODE REVIEW

File: app/api/events.py [hunk 1]
```diff
@@ -287,7 +287,7 @@ def after_create_object(self, event, data, view_kwargs):
 class EventDetail(ResourceDetail):
 
     def before_get_object(self, view_kwargs):
-        if view_kwargs.get('identifier'):
+        if view_kwargs.get('event_identifier'):
```
Comment (shubham-padia) [inline app/api/events.py]: Please make the same change on line 382

Comment (enigmaeth) [inline app/api/events.py]: Resolved. I am looking for other errors that might have crept in after the `safe_query` addition. 
@shubham-padia Please check for yourself too :smile:

File: app/api/events.py [hunk 2]
```diff
@@ -380,8 +380,8 @@ def before_get_object(self, view_kwargs):
 class EventRelationship(ResourceRelationship):
 
     def before_get_object(self, view_kwargs):
-        if view_kwargs.get('identifier'):
-            event = self.session.query(Event).filter_by(identifier=view_kwargs['identifier']).one()
```
Comment (shubham-padia) [inline app/api/events.py]: @enigmaeth, can you use safe_query here, looks like I forgot it here :sweat_smile:

Comment (enigmaeth) [inline app/api/events.py]: Resolved.

File: app/api/events.py [hunk 3]
```diff
@@ -381,7 +381,7 @@ class EventRelationship(ResourceRelationship):
 
     def before_get_object(self, view_kwargs):
         if view_kwargs.get('identifier'):
-            event = self.session.query(Event).filter_by(identifier=view_kwargs['identifier']).one()
+            event = safe_query(self, Event, 'identifier', view_kwargs['identifier'], 'identifier')
             view_kwargs['id'] = event.id
```
Comment (shubham-padia) [inline app/api/events.py]: Please use `event_identifier` in the above three lines

Comment (enigmaeth) [inline app/api/events.py]: Well, then we have to modify all the relationship endpoints in __init__.py. I thought we can make those changes in another issue.

File: app/api/__init__.py [hunk 4]
```diff
@@ -75,8 +75,9 @@
 api.route(EventList, 'event_list', '/events', '/event-types/<int:event_type_id>/events',
           '/event-topics/<int:event_topic_id>/events',
           '/event-sub-topics/<int:event_sub_topic_id>/events')
-api.route(EventDetail, 'event_detail', '/events/<int:id>', '/events/<identifier>', '/tickets/<int:ticket_id>/event',
```
Comment (shubham-padia) [inline app/api/__init__.py]: Revert this change

File: app/api/events.py (hunk 1 above)
Comment (shubham-padia) [inline app/api/events.py]: change this to 
```python
if view_kwargs.get('identifier'):
event = safe_query(self, Event, 'identifer', view_kwargs['identifier'], 'identifier')
```

File: app/api/events.py (hunk 3 above)
Comment (shubham-padia) [inline app/api/events.py]: Keep this the same now

Comment (shubham-padia) [inline app/api/events.py]: see the new review, reiterate according to that.

**Comment by niranjan94** [comment]
@enigmaeth The base branch history was modified. Please update your branch and fix your commits.

**Comment by enigmaeth** [comment]
@niranjan94 @SaptakS @shubham-padia  Updated. Please review.
